### PR TITLE
Show EVM block number when the user is on EVM chain

### PR DESCRIPTION
**Pull request by RodionChachura**
image

* It works only for users with an EVM wallet.
* After switching to the Avalanche network, it can take time to get the web3 provider, so the block number could be missing for a few seconds.

### CODE REVIEW

File: app/src/@anchor-protocol/app-provider/queries/evm/evmBlockNumber.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  const result = useQuery(
+    [ANCHOR_QUERY_KEY.EVM_BLOCK_NUMBER, provider?.network?.chainId],
+    async () => {
+      const blockNumber = await provider?.getBlockNumber();
+      console.log(provider);
+      return blockNumber;
+    },
+    {
+      refetchInterval: 1000 * 60,
+      keepPreviousData: true,
```
Comment (cosullivan) [inline app/src/@anchor-protocol/app-provider/queries/evm/evmBlockNumber.ts:19]: can set this to false as we dont need to ever fetch old values from this query.

File: app/src/components/Footer/index.tsx [hunk 2]
```diff
@@ -43,16 +50,20 @@ function FooterBase({ className, style }: FooterProps) {
...
+          <BlockInfo
+            chainName={Chain.Terra}
+            networkName={network.name}
+            blockNumber={lastSyncedHeight}
+          />
         </a>
 
-        <Link to="/terms">Terms</Link>
+        {isEVM && evmBlockNumber && (
+          <BlockInfo blockNumber={evmBlockNumber} chainName={chain} />
```
Comment (cosullivan) [inline app/src/components/Footer/index.tsx:59]: I think in this case we should create an EvmBlockInfo component which just returns BlockInfo but also encapsulates the useEvmBlockNumber hook.. this way we can stop the EVM block number query from occurring on the terra network as the hook will only run when that EvmBlockInfo component is executed.

File: app/src/@anchor-protocol/app-provider/queries/evm/evmBlockNumber.ts (hunk 1 above)
Comment (RodionChachura) [reply app/src/@anchor-protocol/app-provider/queries/evm/evmBlockNumber.ts]: ✅ https://github.com/Anchor-Protocol/anchor-web-app/pull/304/commits/2ab64bdbb4da953bb9c3ba77cff867a31f94586e

File: app/src/components/Footer/index.tsx (hunk 2 above)
Comment (RodionChachura) [reply app/src/components/Footer/index.tsx]: ✅ https://github.com/Anchor-Protocol/anchor-web-app/pull/304/commits/cbe77e40d8974f85295f0ddf5352fbcf09671a69

**Pull request merged**